adapters/brave/brave.go

```diff
@@ -0,0 +1,164 @@
+package brave
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the Brave adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	uri, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: uri,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var braveExt *openrtb_ext.ExtImpBrave
+	var err error
+
+	for i, imp := range request.Imp {
+		braveExt, err = a.getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			break
+		}
+		request.Imp[i].Ext = nil
+	}
```

Comment:
With this code, the braveExt from the last imp is the only one which matters. It looks like your server includes the placement id in the url and thus only supports one placement per request? If so, please consider impression splitting: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#impression-splitting

Reply:
yes guys you right, removed it as not required (first one rewrite on loop not make sense, also we using only one impression array index on backends)